File: HabitFormationApp/src/HabitFormationApp.Core/Model/HabitAggregate/Habit.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace HabitFormationApp.Core;

public class Habit
{
    public Guid HabitId { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public HabitFrequency Frequency { get; set; }
    public int TargetDaysPerWeek { get; set; } = 7;
    public DateTime StartDate { get; set; } = DateTime.UtcNow;
    public bool IsActive { get; set; } = true;
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<Streak> Streaks { get; set; } = new List<Streak>();
    
    public void ToggleActive()
    {
        IsActive = !IsActive;
    }
}

File: HabitFormationApp/src/HabitFormationApp.WebApp/projects/habit-formation-app/src/app/models/habit.ts
import { HabitFrequency } from './habit-frequency';

export interface Habit {
  habitId: string;
  userId: string;
  name: string;
  description?: string;
  frequency: HabitFrequency;
  targetDaysPerWeek: number;
  startDate: string;
  isActive: boolean;
  notes?: string;
  createdAt: string;
}

export interface CreateHabitRequest {
  userId: string;
  name: string;
  description?: string;
  frequency: HabitFrequency;
  targetDaysPerWeek: number;
  startDate: string;
  notes?: string;
}

export interface UpdateHabitRequest {
  habitId: string;
  name: string;
  description?: string;
  frequency: HabitFrequency;
  targetDaysPerWeek: number;
  notes?: string;
}
